### DISCUSSION TITLE
how to suspend services

### DISCUSSION THREAD

**Discussion by fcomte** [InseeFrLab/onyxia#791] [Ideas]
As the usage of Onyxia increases, the demand for cleaning services becomes crucial. We are interested in the ability to suspend services rather than completely terminating them, to avoid the risk of losing user data.

The method for detecting when a service is currently not in use falls outside the scope of this conversation, as we anticipate it might not be integrated into Onyxia, at least in the short term. 

This functionality is not presently available in the Helm ecosystem, and we must decide a plan for its implementation.

Initially, we must determine:

- Whether the chart itself should be responsible for this functionality, for example, by setting a specific global value such as `global.suspend = true | false`.
- Whether we should identify all replicas of all stateful sets (STS) and deployments to adjust their count to 0.

We are inclined to explore the first option as it is simpler and assigns the responsibility directly to the chart.

In our SSPCloud platform, by enabling the suspension of services (and allowing users to resume them), we can quickly suspend services without concern for data loss.

**Comment by olevitt** [comment]
Implementation for solution 1 `global.suspend` is done API side : https://github.com/InseeFrLab/onyxia-api/pull/413 (needs some refining before merging but mostly works)  
Work on UI side has been defined here : https://github.com/InseeFrLab/onyxia/issues/793  
This is what it currently looks like to implement in a service :  
Add the `global.suspend` property to the `values.schema.json` (this is what sets the service as `pausable` in regards to Onyxia) : e.g https://github.com/InseeFrLab/helm-charts-dev/commit/968c7f94a9f2f8c526ed8bfb6a2cd9562ad8bddf
Use the `global.suspend` value wherever you think it makes sense : e.g https://github.com/InseeFrLab/helm-charts-dev/blob/c9a9e160777931d6742f24d946f386e92e545129/charts/vscode-python/templates/statefulset.yaml#L10

We will have to document this behaviour for users, admins and catalogs maintainers
